### ISSUE TITLE
Windows installer should launch the Pd-GUI (rather than the Pd-DSP)

### ISSUE THREAD

**Issue by umlaeute** [pure-data/pure-data#480]
as mentioned in #479 there is some regression in how Pd is launched on windows.

The Pd-GUI (the tcl/tk interpreter running `pd-gui.tcl`) has some singleton-protection to avoid being doubly-started. This protection only kicks in, when the Pd-GUI is  being started directly (and in turn starts the Pd-DSP).
It does not do anything, when Pd-GUI is being started (indirectly) via the Pd-DSP.

As such, i think that the nsis-Installer should launch the Pd-GUI rather than the Pd-DSP.
(so it works consistently with e.g. the Debian package; i don't know about the current state on macOS)

**Comment by umlaeute** [comment]
mentioning @Lucarda, since you gave some love to the NSIS-installer recently and might fix this as well.

**Comment by umlaeute** [comment]
the association should be something like:  `"$INSTDIR\bin\wish86.exe $INSTDIR\tcl\pd-gui.tcl -open %1"`

the `wish86.exe` probably needs to be computed when the nsi-script is updated (to allow for other tcl/tk interpreters

**Comment by Lucarda** [comment]
Right.
this was impossible until #481

The NSIS thing should somehow end with this output on the registry:
`"C:\Program Files (x86)\Pd\bin\wish85.exe" "C:\Program Files (x86)\Pd\tcl\pd-gui.tcl"  "%1"`

Not yet in the nsis installer because I have to solve the .pd icon thing that is now using wish85 icon.

**Comment by umlaeute** [comment]
right. the icon-thing is why i didn't even start (not having a full test-environment at hand)

**Comment by umlaeute** [comment]
just to make sure: with ".pd icon thing", do you only mean the desktop/startmenu icon for starting Pd?
or also the window/taskbar icons once it has launched?

**Comment by Lucarda** [comment]
Only and only the .pd files in the explorer.

![5](https://user-images.githubusercontent.com/25239522/45871179-2d0c6880-bd63-11e8-8032-97fcf634301d.jpg)


I even try to change wish85.exe icon but so far i can't (the registry gets what i send with NSIS):

```
  ;Dummy wish app to display icons in explorer.
    WriteRegStr HKLM "Software\Microsoft\Windows\CurrentVersion\App Paths\wish85.exe" "" ""
  ; Pd icon to wish app
    WriteRegExpandStr HKCR "Applications\wish85.exe\DefaultIcon" "(Default)" "$INSTDIR\tcl\pd.ico"
```

**Comment by umlaeute** [comment]
see my recent pushes to #476

**Comment by millerpuckette** [comment]
merged

**Comment by umlaeute** [comment]
@millerpuckette are you aure you merged? because this issue is closed via #481 which as not been merged.

**Comment by millerpuckette** [comment]
I believed I had merged this but can't find it in my history anywhere - perhaps I failed to actually do it somehow.
